doc/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,422 @@
+                      Its a Metric used to specify preference of next-hop
+                      if this distance is not set, 0 will be set to maintain the set;
+nexthop-vrf         =  1\*15VCHAR ; list of next-hop VRFs. It should be set only if ifname or nexthop IP  is not
+                       in the current VRF . The value is set to VRF name
+                       to which the interface or nexthop IP  belongs for route leaks.
+blackhole           =  List of boolean; true if the next-hop route is blackholed.                     
+```
+The nexthop-vrf if set, will allow to create a leaked route in the current VRF(vrf-name). The vrf-name, nexthop-vrf, prefix, ifname, distance and blackhole are all parameters required to configure static routes and route leaks in vtysh.
+If an interface is moved from one VRF to another and it exist in the STATIC_ROUTE table, error will be returned by the backend. User is expected to remove all static routes pertaining to the interface before binding the interface to a different VRF.
+If prefix is IPv4Prefix nexthop will have  IPv4Address and if  prefix is IPv6Prefix nexthop will have  IPv6Address.
+In this table each entry based on the index in the lists of  nexthop, ifname, distance, nexthop-vrf and blackhole  is a set. Together based on the index they specify one next-hop entry as there can be multiple next-hops for a prefix. Empty string will be used to complete the set if required.
+
+In the example -1 table below (0.0.0.0, Ethernet0, 10, default, false) , (2.2.2.2, "", 0,"", false), (0.0.0.0, Ethernet12, 30, Vrf-GREEN, false) are the 3 sets defined for 3 configured next-hops for prefix 10.11.1.0/24 in Vrf-RED. 
+In example -2  the blackhole is true for second index and  only distance is relevant in blackholed route. Due to blackhole entry (with metric 40) all packets matching 10.12.11.0/24 will be discarded if other specified  nexthop is inactive.
+
+```JSON
+Example -1 :--
+
+key                 = STATIC_ROUTE|Vrf-RED|10.11.1.0/24;
+vrf-name            = Vrf-RED
+prefix              = 10.11.1.0/24
+nexthop             = 0.0.0.0, 2.2.2.1, 0.0.0.0
+ifname              = "Ethernet0", "", "Ethernet12" 
+distance            = 10,0,30
+nexthop-vrf         = "default", "" , "Vrf-GREEN"
+blackhole           = false, false, false
+
+EXAMPLE 2:-
+key                 = STATIC_ROUTE|default|10.12.11.0/24;
+vrf-name            = default
+prefix              = 10.12.11.0/24
+nexthop             = 2.2.2.3, ""
+ifname              = "", ""
+distance            = 10,40
+nexthop-vrf         = "",""
+blackhole           = false, true
+
+
+```
+
+
+Note: This model is proposed in line with ROUTE_TABLE model in application DB used for route updates and "vrf hld" description of STATIC_ROUTE table (which is currently not in use). Some default values like 0.0.0.0 and empty strings are used to complete a set. See the CLI example later to see how the table is filled up step-by-step.
+
+
+### 3.2.2 APP DB

```

Comment:
Please remove these empty sections or state, "No changes to APP_DB" etc.

Reply:
ok